### PR TITLE
update Migration guide: info about the changed behavior

### PR DESCRIPTION
**Pull request by ckeshava**
## High Level Overview of Change

This PR includes additional information about the default signing algorithm in the migration guide. We hope this reduces the confusion amongst the library users.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [x] Documentation Updates
- [ ] Release

### Did you update HISTORY.md?

- [ ] Yes
- [ ] No, this change does not impact library users
This PR does not have any material code changes

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -234,6 +234,10 @@ This was done to remove a hard dependency on `https-proxy-agent` when running
 
 `Wallet.generate()` and `Wallet.fromSeed` were ignoring the `algorithm` parameter. This means that if you were manually specifying `algorithm` in any `Wallet` constructors, you may generate a different `Wallet` keypair when upgrading to 3.0. In that case to get the same generated wallets as before, don’t specify the `algorithm` parameter.
 
+NOTE: THE DEFAULT SIGNING ALGORITHM USED IN `Wallet.generate()` IS SECP256K1. If you do not specify any `algorithm` parameter, a `secp256k1` seed is generated. If you need the seed to be compliant with the `ed25519` signing algorithm, please use `algorithm=ed25519` parameter.
```
Comment (mvadari) [inline MIGRATION.md:237]: This should be bolded

File: MIGRATION.md [hunk 2]
```diff
@@ -234,6 +234,10 @@ This was done to remove a hard dependency on `https-proxy-agent` when running
...
+
+This behavior is expected to change in future releases of xrpl.js. We intend to make `ed25519` as the default signing algorithm in `Wallet.generate()` function. This would improve compatibility with the `Wallet.fromSeed` method, because it's beneficial to have identical defaults in both the functions.
```
Comment (mvadari) [inline MIGRATION.md:239]: This is unnecessary here IMO.

Comment (ckeshava) [reply MIGRATION.md]: I made significant changes to the wording of the migration guide. Please let me know if that's alright

**Comment by ckeshava** [comment]
thnks @JST5000  for helping me re-word the migration guide. let me know if this looks alright ^^

**Review by justinr1234 — APPROVED** [review]

File: MIGRATION.md [hunk 3]
```diff
@@ -230,9 +233,27 @@ This was done to remove a hard dependency on `https-proxy-agent` when running
...
+
+If you are creating new accounts each time (ex. via `Client.fundWallet`), you do not need to specify the signing algorithm.
+
+**Before**
+
+`Wallet.fromSeed('s...')`
+`Wallet.generate()`
+`Wallet.fromEntropy(entropy)`
+`deriveKeyPair(seed="s...")`
+`generateSeed()`
```
Comment (mvadari) [inline MIGRATION.md:248]: This should be one code block to make it easier to read

File: MIGRATION.md [hunk 4]
```diff
@@ -230,9 +233,27 @@ This was done to remove a hard dependency on `https-proxy-agent` when running
...
+
+**After**
 
-`Wallet.generate()` and `Wallet.fromSeed` were ignoring the `algorithm` parameter. This means that if you were manually specifying `algorithm` in any `Wallet` constructors, you may generate a different `Wallet` keypair when upgrading to 3.0. In that case to get the same generated wallets as before, don’t specify the `algorithm` parameter.
+`Wallet.fromSeed(seed='s...',algorithm: 'ecdsa-secp256k1')`
+`Wallet.generate(algorithm: 'ecdsa-secp256k1')`
+`Wallet.fromEntropy(entropy, opts={algorithm: 'ecdsa-secp256k1'})`
+`deriveKeypair(seed='s...', opts={ algorithm: 'ecdsa-secp256k1' }) (ripple-keypairs)`
+`generateSeed({ entropy, algorithm: 'ecdsa-secp256k1' } (ripple-keypairs)`
```
Comment (mvadari) [inline MIGRATION.md:256]: ditto here

File: MIGRATION.md [hunk 5]
```diff
@@ -230,9 +233,27 @@ This was done to remove a hard dependency on `https-proxy-agent` when running
    authorization: 'authorization'
  }`
 
-### 8. Bug fix: Setting an explicit `algorithm` when generating a wallet works now
+### 8. `Wallet` functions default to `ed25519` instead of `secp256k1` signing algorithm
+
+With 3.0, we updated the default signing algorithm used by the Wallet object from secp256k1 to ed25519 in order to default to the higher-performance algorithm. This is a breaking change to all functions used to generate a Wallet, so if you have a pre-existing XRPL account that you're using to generate a specific Wallet using older versions of xrpl.js, you must specify that you are using secp256k1 as the algorithm to decode your private key / seed / etc to get the same behavior as before. See below for specifically how to update your code.
+
+If you are creating new accounts each time (ex. via `Client.fundWallet`), you do not need to specify the signing algorithm.
```
Comment (mvadari) [inline MIGRATION.md:240]:
```suggestion
If you are creating new accounts each time (ex. via `Client.fundWallet` or `Wallet.generate`), you do not need to specify the signing algorithm.
```

File: MIGRATION.md [hunk 6]
```diff
@@ -230,9 +233,27 @@ This was done to remove a hard dependency on `https-proxy-agent` when running
    authorization: 'authorization'
  }`
 
-### 8. Bug fix: Setting an explicit `algorithm` when generating a wallet works now
+### 8. `Wallet` functions default to `ed25519` instead of `secp256k1` signing algorithm
+
+With 3.0, we updated the default signing algorithm used by the Wallet object from secp256k1 to ed25519 in order to default to the higher-performance algorithm. This is a breaking change to all functions used to generate a Wallet, so if you have a pre-existing XRPL account that you're using to generate a specific Wallet using older versions of xrpl.js, you must specify that you are using secp256k1 as the algorithm to decode your private key / seed / etc to get the same behavior as before. See below for specifically how to update your code.
```
Comment (mvadari) [inline MIGRATION.md:238]:
```suggestion
With 3.0, we updated the default signing algorithm used by the Wallet object from secp256k1 to ed25519 in order to default to the higher-performance algorithm. This is a breaking change to all functions used to generate a Wallet, so if you have a pre-existing XRPL account that you're using to generate a specific Wallet using older versions of xrpl.js, you may need to specify that you are using secp256k1 as the algorithm to decode your private key / seed / etc to get the same behavior as before. See below for specifically how to update your code.
```

File: MIGRATION.md [hunk 7]
```diff
@@ -230,9 +233,27 @@ This was done to remove a hard dependency on `https-proxy-agent` when running
...
+
+If you are creating new accounts each time (ex. via `Client.fundWallet`), you do not need to specify the signing algorithm.
+
+**Before**
+
+`Wallet.fromSeed('s...')`
+`Wallet.generate()`
```
Comment (mvadari) [inline MIGRATION.md:245]: `Wallet.generate` users don't need to change.

File: MIGRATION.md (hunk 3 above)
Comment (mvadari) [inline MIGRATION.md:248]: `generateSeed` don't need to change.

File: MIGRATION.md (hunk 7 above)
Comment (ckeshava) [reply MIGRATION.md:245]: yes, although the internals of the `Wallet.generate` has been updated to incorporate the `algorithm` parameter (PR: https://github.com/XRPLF/xrpl.js/commit/041055082a6f689ea30dcc1e1a04aba6103f537b), the external behavior hasn't changed (as evidenced by the un-modified `Wallet.generate` unit tests)

File: MIGRATION.md (hunk 3 above)
Comment (ckeshava) [reply MIGRATION.md:248]: if no explicit `algorithm` parameter is provided, `generateSeed` produces a seed of type `secp256k1`. Here's an example unit test: https://github.com/XRPLF/xrpl.js/blob/be732a4a6b7bdf58d0efff084b76a66f3f625fe6/packages/ripple-keypairs/test/api.test.ts#L26

I feel this behavior contradicts the spirit of the update: The default behavior of `generateSeed` function is to produce a `secp256k1` seed, not a `ed25519` seed.

What do you suggest? How can we communicate this?

Comment (ckeshava) [reply MIGRATION.md:248]: this change could fix this issue: https://github.com/XRPLF/xrpl.js/pull/2658/files#diff-37199a4e4bdd9fb657c619a808e98350642688e38494ad56c36372b43c6437b8L19

but it needs some more time and review

File: MIGRATION.md (hunk 4 above)
Comment (ckeshava) [reply MIGRATION.md]: fixed in [e216200](https://github.com/XRPLF/xrpl.js/pull/2659/commits/e216200ec2a94a39939dba55039bc5bedb389ed7)

File: MIGRATION.md (hunk 3 above)
Comment (ckeshava) [reply MIGRATION.md]: fixed in [e216200](https://github.com/XRPLF/xrpl.js/pull/2659/commits/e216200ec2a94a39939dba55039bc5bedb389ed7)

Comment (mvadari) [reply MIGRATION.md]: That's fair, but you're still usually not providing entropy.

File: MIGRATION.md [hunk 8]
```diff
@@ -2,7 +2,10 @@
...
+The two main changes you'll notice are: a breaking change to `Wallet` object creation, to use a more performanant algorithm by default and the update replacing `Buffer` with `Uint8Array` across the board. This was done since browsers don't support `Buffer`. Fortunately, this transition is relatively straightforward, as `Buffer` is a subclass of `Uint8Array`, meaning in many circumstances `Buffer` can be directly replaced by `Uint8Array`. The primary difference is that `Buffer` has additional helper functions. We've listed the affected client library functions below in the `Uint8Array` section for your reference.
+
+
+**Creation of a `Wallet` has breaking changes. See **[Link](#8-wallet-functions-default-to-ed25519-instead-of-secp256k1-signing-algorithm) for details**
```
Comment (mvadari) [inline MIGRATION.md:8]:
```suggestion
The two main changes you'll notice are:
* A breaking change to `Wallet` object creation, to use a more performant algorithm by default. See [Link](#8-wallet-functions-default-to-ed25519-instead-of-secp256k1-signing-algorithm) for details.
* Replacing `Buffer` with `Uint8Array` across the board. This was done since browsers don't support `Buffer`. Fortunately, this transition is relatively straightforward, as `Buffer` is a subclass of `Uint8Array`, meaning in many circumstances `Buffer` can be directly replaced by `Uint8Array`. The primary difference is that `Buffer` has additional helper functions. We've listed the affected client library functions below in the `Uint8Array` section for your reference.
```

File: MIGRATION.md [hunk 9]
```diff
@@ -230,9 +233,27 @@ This was done to remove a hard dependency on `https-proxy-agent` when running
...
+### 8. `Wallet` functions default to `ed25519` instead of `secp256k1` signing algorithm
+
+With 3.0, we updated the default signing algorithm used by the Wallet object from secp256k1 to ed25519 in order to default to the higher-performance algorithm. This is a breaking change to all functions used to generate a Wallet, so if you have a pre-existing XRPL account that you're using to generate a specific Wallet using older versions of xrpl.js, you may need to specify that you are using secp256k1 as the algorithm to decode your private key / seed / etc to get the same behavior as before. See below for specifically how to update your code.
+
+If you are creating new accounts each time (ex. via `Client.fundWallet` or `Wallet.generate`), you do not need to specify the signing algorithm.
 
-`Wallet.generate()` and `Wallet.fromSeed` were ignoring the `algorithm` parameter. This means that if you were manually specifying `algorithm` in any `Wallet` constructors, you may generate a different `Wallet` keypair when upgrading to 3.0. In that case to get the same generated wallets as before, don’t specify the `algorithm` parameter.
```
Comment (mvadari) [inline MIGRATION.md:235]: Some form of this text should still be included in this section, to explain why the change happened. Otherwise it seems random.

Comment (ckeshava) [reply MIGRATION.md:235]: okay, updated in [a16773e](https://github.com/XRPLF/xrpl.js/pull/2659/commits/a16773ee21847ed903fca9f0c6b77098890201fb)

File: MIGRATION.md (hunk 3 above)
Comment (ckeshava) [reply MIGRATION.md]: okay, fixed in [a16773e](https://github.com/XRPLF/xrpl.js/pull/2659/commits/a16773ee21847ed903fca9f0c6b77098890201fb)

File: MIGRATION.md (hunk 7 above)
Comment (ckeshava) [reply MIGRATION.md]: fixed in [a16773e](https://github.com/XRPLF/xrpl.js/pull/2659/commits/a16773ee21847ed903fca9f0c6b77098890201fb)

File: packages/ripple-keypairs/src/index.ts [hunk 10]
```diff
@@ -38,7 +38,7 @@ function generateSeed(
   const entropy = options.entropy
     ? options.entropy.slice(0, 16)
     : randomBytes(16)
-  const type = options.algorithm === 'ed25519' ? 'ed25519' : 'secp256k1'
+  const type = options.algorithm === 'ecdsa-secp256k1' ? 'secp256k1' : 'ed25519'
```
Comment (mvadari) [inline packages/ripple-keypairs/src/index.ts:41]: Why is this behavior changed in this PR? That should be in a separate PR.

Comment (ckeshava) [reply packages/ripple-keypairs/src/index.ts]: thanks. I had pushed these changed by mistake

Comment (ckeshava) [reply packages/ripple-keypairs/src/index.ts]: fixed in [dfb14d3](https://github.com/XRPLF/xrpl.js/pull/2659/commits/dfb14d3eab2b9f1bd1dba9eb9259fed9f804f0b0)

**Review by JST5000 — APPROVED** [review]
LGTM

**Review by justinr1234 — APPROVED** [review]

**Comment by ckeshava** [comment]
@mvadari  do the changes look okay? I believe I have addressed your comments

File: MIGRATION.md [hunk 11]
```diff
@@ -2,7 +2,9 @@
 
 In xrpl.js 3.0, we've made significant improvements that result in a 60% reduction in bundle size for browser applications. We've also eliminated the need for polyfills with minimal disruption to existing code. This was achieved by replacing node-specific dependencies with ones that are compatible with browsers.
 
-The main change you'll notice is the update replacing `Buffer` with `Uint8Array` across the board. This was done since browsers don't support `Buffer`. Fortunately, this transition is relatively straightforward, as `Buffer` is a subclass of `Uint8Array`, meaning in many circumstances `Buffer` can be directly replaced by `Uint8Array`. The primary difference is that `Buffer` has additional helper functions. We've listed the affected client library functions below in the `Uint8Array` section for your reference.
+The two main changes you'll notice are:
+* A breaking change to `Wallet` object creation, to use a more performant algorithm by default. See [Link](#8-wallet-functions-default-to-ed25519-instead-of-secp256k1-signing-algorithm) for details.
```
Comment (mvadari) [inline MIGRATION.md:6]: nit:
```suggestion
* A breaking change to `Wallet` object creation, to use a more performant algorithm by default. See [here](#8-wallet-functions-default-to-ed25519-instead-of-secp256k1-signing-algorithm) for details.
```

File: MIGRATION.md [hunk 12]
```diff
@@ -230,9 +232,27 @@ This was done to remove a hard dependency on `https-proxy-agent` when running
...
+In previous releases of this library, `Wallet.generate()` and `Wallet.fromSeed` were ignoring the `algorithm` parameter. Instead, the algorithm was assumed from the seed provided; if it started with `sEd`, it would use `ed25519`, and otherwise it would use `secp256k1`. However, seeds do not actually have algorithms; a seed starting with `s...` can still use the `ed25519` algorithm.
+
+With 3.0, we updated the default signing algorithm used by the `Wallet` object to always be `ed25519` in order to default to the higher-performance algorithm. This is a breaking change to all functions used to generate a Wallet, so if you have a pre-existing XRPL account that you're using to generate a specific Wallet using older versions of xrpl.js, you may need to specify that you are using secp256k1 as the algorithm to decode your private key / seed / etc to get the same behavior as before. See below for specifically how to update your code.
```
Comment (mvadari) [inline MIGRATION.md:239]: nit:
```suggestion
With 3.0, we updated the default signing algorithm used by the `Wallet` object to always be `ed25519` in order to default to the higher-performance algorithm. This is a breaking change to all functions used to generate a Wallet, so if you have a pre-existing XRPL account that you're using to generate a specific Wallet using older versions of xrpl.js, you may need to specify that you are using `secp256k1` as the algorithm to decode your private key / seed / etc to get the same behavior as before. See below for specifically how to update your code.
```

**Review by mvadari — APPROVED** [review]
